eva-accession-pipeline/src/main/java/uk/ac/ebi/eva/accession/pipeline/steps/tasklets/reportCheck/ReportCheckTasklet.java

```diff
@@ -0,0 +1,201 @@
+import org.springframework.batch.item.ItemStreamReader;
+import org.springframework.batch.repeat.RepeatStatus;
+
+import uk.ac.ebi.eva.accession.pipeline.policies.InvalidVariantSkipPolicy;
+import uk.ac.ebi.eva.commons.core.models.pipeline.Variant;
+
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Compares the the original VCF and accession report VCF, and logs the differences.
+ *
+ * Terminology: The original VCF contains variants, and it will be loaded into the 'variant buffer'. The accession
+ * report VCF contains the accessioned variants, and it will be loaded into the 'accession buffer'.
+ *
+ * The high-level idea of the process is filling those 2 buffers, and when the same coordinates appears in both the
+ * variant buffer and the accession buffer, it means that the variant in those coordinates was correctly accessioned,
+ * so it can be removed from both buffers. At the end, only variants without accessions and accessions without
+ * variants should be left in the buffers after both files were completely read.
+ *
+ * The reason why the buffers are needed is that the accession report can be unordered. In order to avoid
+ * having the VCFs completely loaded in memory, a buffer size can be configured for both buffers. However, to guarantee
+ * correct behaviour in worst-ordering scenario, the buffers need to be able to grow indefinitely.
+ *
+ * To perform some self-checks, this tasklet provides the maximum size of the buffers during the execution, and
+ * also provides the number of iterations needed.
+ */
+public class ReportCheckTasklet implements Tasklet {
+
+    private static final Logger logger = LoggerFactory.getLogger(ReportCheckTasklet.class);
+
+    private static final int DEFAULT_MAX_BUFFER_SIZE = 100000;
+
+    private static final int BUFFER_SIZE_INCREASE_FACTOR = 2;
+
+    private ItemStreamReader<Variant> inputReader;
+
+    private ItemStreamReader<Variant> reportReader;
+
+    private long maxBufferSize;
+
+    private long initialBufferSize;
+
+    private long iterations;
+
+    private SkipPolicy skipPolicy;
+
+    public ReportCheckTasklet(ItemStreamReader<Variant> inputReader, ItemStreamReader<Variant> reportReader) {
+        this.inputReader = inputReader;
+        this.reportReader = reportReader;
+        this.initialBufferSize = DEFAULT_MAX_BUFFER_SIZE;
+        this.maxBufferSize = 0;
+        this.iterations = 0;
+        this.skipPolicy = new InvalidVariantSkipPolicy();
+    }
+
+    @Override
+    public RepeatStatus execute(StepContribution stepContribution, ChunkContext chunkContext) throws Exception {
+        Set<Variant> variantBuffer = new HashSet<>();
+        Set<Variant> accessionBuffer = new HashSet<>();
+        inputReader.open(new ExecutionContext());
+        reportReader.open(new ExecutionContext());
+        boolean finishedInputFile = false;
+        boolean finishedReportFile = false;
+        while (!(finishedInputFile && finishedReportFile)) {
```

Comment:
This condition could be simplified if the meaning of the variables was inverted, as in `pendingInputFile`:

```java
boolean pendingInputFile = true;
boolean pendingReportFile = true;
while (pendingInputFile || pendingReportFile) {
    // ...
}
```

The return value in `fillbuffer` would then be `true` if more data can be read.

I don't have a strong opinion on this though.

Reply:
I like a little bit more "while (pendingInputFile || pendingReportFile)"
In the other hand, if I see that fillBuffer() returns true, I tend to think more that it has finished. But the variable name in the side will clarify that. Looking one or the other you know what fillBuffer is returning:
finishedInputFile = fillBuffer(inputReader, variantBuffer);
vs 
pendingInputFile = fillBuffer(inputReader, variantBuffer);

Reply:
I would change it then, but is OK also like it is